geopandas/geoseries.py

```diff
@@ -389,22 +390,38 @@ def to_crs(self, crs=None, epsg=None):
         """
         from fiona.crs import from_epsg
 
+        if crs is None and epsg is None:
+            raise TypeError("Must set either crs or epsg for output.")
+
         if self.crs is None:
             raise ValueError(
                 "Cannot transform naive geometries.  "
                 "Please set a crs on the object first."
             )
+
         if crs is None:
             try:
                 crs = from_epsg(epsg)
```

Comment:
@snowman2 I moved this epsg -> crs conversion back before the exact equality check, in the assumption this is also more correct (eg if you still have a proj4 dict with epsg number in it for `self.crs`, it will only compare equal to an epsg number formatted in the same way).

But actually, we should probably just get rid of this fiona `from_epsg` call I suppose (which just converts the integer number into `{'init': 'epsg:int', 'no_defs': True}`). And rather convert the integer into `EPSG:int`. Or at least do that at some point (maybe when fully switching to using pyproj.CRS objects to store the information)

But I assume that